app/db/daos/document_type_dao.py

```diff
@@ -0,0 +1,21 @@
+from sqlalchemy import select
+
+from app.db.abstract_dao import AbstractDAO
+from app.db.models.document_type import DocumentType
+
+
+class DocumentTypeDAO(AbstractDAO):
+    """
+    Data Access Object for document type
+    """
+
+    async def get_document_type_by_uri(self, uri: str) -> DocumentType | None:
```

Comment:
A "URL" or "URI" type would be more suitable than str, but this may not exist in python